File: app/src/main/java/com/example/administrator/mianshitest/MainActivity.java
package com.example.administrator.mianshitest;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.example.administrator.mianshitest.sqlite.SqliteActivity;

/**
 * User: Lazy_xu
 * Data: 2019/07/02
 * Description:
 * FIXME
 */
public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main1);
        findViewById(R.id.btnSqlite).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, SqliteActivity.class));
            }
        });

    }
}

File: mvvmproject/src/main/java/com/lazyxu/mvvmproject/ui/MainActivity.kt
package com.lazyxu.mvvmproject.ui

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import com.lazyxu.mvvmproject.R
import com.lazyxu.mvvmproject.base.BaseApplication
import com.lazyxu.mvvmproject.data.FilmUseCase
import com.lazyxu.mvvmproject.di.component.DaggerActivityComponent
import com.lazyxu.mvvmproject.utils.LogUtil
import kotlinx.android.synthetic.main.activity_main.*
import javax.inject.Inject

/**
 * User: xuyexiang
 * Date: 2019/06/06
 * Description:
 * FIXME
 */
class MainActivity : AppCompatActivity() {
    @Inject
    lateinit var mineUseCase: FilmUseCase

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        DaggerActivityComponent.builder().appComponent(BaseApplication.getInstance().appComponent).build().inject(this)
        mineUseCase!!.hotFilm().subscribe({ mtimeFilmeEntity -> }, { throwable -> })
        Log.d(LogUtil.TEST_DAGGER, "FilmUseCase地址=" + mineUseCase!!)
//        Log.d(LogUtil.TEST_DAGGER, "FilmUseCase地址=" + mineUseCaseTest!!)
        hideButton.text = "FilmUseCase地址=" + mineUseCase;
    }
}

